modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/RustServerCodegen.java

```diff
@@ -1339,7 +1339,7 @@ public int compare(Map.Entry<String, Map<String, String>> a, Map.Entry<String, M
                 return a.getValue().get("path").compareTo(b.getValue().get("path"));
             }
         });
-        List pathSet = new ArrayList<Map<String, String>>();
+        List pathSet = new ArrayList<>();
```

Comment:
this is wrong now, correctly specify set of the list

Reply:
That's interesting. Compiles for me and pull request checks did pass, too. What do you think is "more wrong" now?

Reply:
@agilob Can you please elaborate on "introduce regressions in use of the type-system" so I have a chance to improve?

Reply:
List pathSet = new ArrayList<Map<String, String>>(); 

this produces List<Map<String,String>>

             List pathSet = new ArrayList<>(); 

this infers List<Object>

Reply:
No.
Java uses type erase for its generics. 
Both new expressions will create the instances of the same class: raw ArrayList

From https://docs.oracle.com/javase/tutorial/java/generics/erasure.html#:~:text=To%20implement%20generics%2C%20the%20Java,classes%2C%20interfaces%2C%20and%20methods.

> Type Erasure
Generics were introduced to the Java language to provide tighter type checks at compile time and to support generic programming. To implement generics, the Java compiler applies type erasure to:

> Replace all type parameters in generic types with their bounds or Object if the type parameters are unbounded. The produced bytecode, therefore, contains only ordinary classes, interfaces, and methods.
> Insert type casts if necessary to preserve type safety.
> Generate bridge methods to preserve polymorphism in extended generic types.
> Type erasure ensures that no new classes are created for parameterized types; consequently, generics incur no runtime overhead.

Demo1.java:
```
import java.util.List;
import java.util.ArrayList;
import java.util.Map;

public class Demo1 {

    public static void main(String[] args) {
        List pathSet = new ArrayList<Map<String, String>>(); 

        pathSet.add("a string");
        System.out.println("size: " + pathSet.size());
        System.out.println("data: " + pathSet);
    }
}
```

Demo2.java:
```
import java.util.List;
import java.util.ArrayList;
import java.util.Map;

public class Demo2 {

    public static void main(String[] args) {
        List pathSet = new ArrayList<>(); 

        pathSet.add("a string");
        System.out.println("size: " + pathSet.size());
        System.out.println("data: " + pathSet);
    }
}
```

Both will run and print the same:
```
size: 1
data: [a string]
```

The byte code:
Demo1:
```
$ javap -c Demo1
Compiled from "Demo1.java"
public class Demo1 {
  public Demo1();
    Code:
       0: aload_0
       1: invokespecial #1                  // Method java/lang/Object."<init>":()V
       4: return

  public static void main(java.lang.String[]);
    Code:
       0: new           #2                  // class java/util/ArrayList
       3: dup
       4: invokespecial #3                  // Method java/util/ArrayList."<init>":()V
       7: astore_1
       8: aload_1
       9: ldc           #4                  // String a string
      11: invokeinterface #5,  2            // InterfaceMethod java/util/List.add:(Ljava/lang/Object;)Z
      16: pop
      17: getstatic     #6                  // Field java/lang/System.out:Ljava/io/PrintStream;
      20: aload_1
      21: invokeinterface #7,  1            // InterfaceMethod java/util/List.size:()I
      26: invokedynamic #8,  0              // InvokeDynamic #0:makeConcatWithConstants:(I)Ljava/lang/String;
      31: invokevirtual #9                  // Method java/io/PrintStream.println:(Ljava/lang/String;)V
      34: getstatic     #6                  // Field java/lang/System.out:Ljava/io/PrintStream;
      37: aload_1
      38: invokedynamic #10,  0             // InvokeDynamic #1:makeConcatWithConstants:(Ljava/util/List;)Ljava/lang/String;
      43: invokevirtual #9                  // Method java/io/PrintStream.println:(Ljava/lang/String;)V
      46: return
}
```

```
$ javap -c Demo2
Compiled from "Demo2.java"
public class Demo2 {
  public Demo2();
    Code:
       0: aload_0
       1: invokespecial #1                  // Method java/lang/Object."<init>":()V
       4: return

  public static void main(java.lang.String[]);
    Code:
       0: new           #2                  // class java/util/ArrayList
       3: dup
       4: invokespecial #3                  // Method java/util/ArrayList."<init>":()V
       7: astore_1
       8: aload_1
       9: ldc           #4                  // String a string
      11: invokeinterface #5,  2            // InterfaceMethod java/util/List.add:(Ljava/lang/Object;)Z
      16: pop
      17: getstatic     #6                  // Field java/lang/System.out:Ljava/io/PrintStream;
      20: aload_1
      21: invokeinterface #7,  1            // InterfaceMethod java/util/List.size:()I
      26: invokedynamic #8,  0              // InvokeDynamic #0:makeConcatWithConstants:(I)Ljava/lang/String;
      31: invokevirtual #9                  // Method java/io/PrintStream.println:(Ljava/lang/String;)V
      34: getstatic     #6                  // Field java/lang/System.out:Ljava/io/PrintStream;
      37: aload_1
      38: invokedynamic #10,  0             // InvokeDynamic #1:makeConcatWithConstants:(Ljava/util/List;)Ljava/lang/String;
      43: invokevirtual #9                  // Method java/io/PrintStream.println:(Ljava/lang/String;)V
      46: return
}
``` 

```
$ diff -u <(javap -c Demo1) <(javap -c Demo2)
--- /dev/fd/63	2021-01-23 01:18:18.506290410 +0100
+++ /dev/fd/62	2021-01-23 01:18:18.506290410 +0100
@@ -1,6 +1,6 @@
-Compiled from "Demo1.java"
-public class Demo1 {
-  public Demo1();
+Compiled from "Demo2.java"
+public class Demo2 {
+  public Demo2();
     Code:
        0: aload_0
        1: invokespecial #1                  // Method java/lang/Object."<init>":()V
```

Reply:
I didn't mean internals of JVM and type erasure, but lose of information caused by use of raw `list` and `arraylist`. This line is removing information what type the list (called "set" btw) should contain. Right now this line and a few others where you did similar changes sells me information that the list must contain `Map<String, String>` after removal of this information I wouldn't know what the list contains.

Later on the list is put as a value into `Map<String, Object> bundle` where we lose more information about type of the map.entry value. I really think if you are doing this refactoring it should provide API users with more information and some more valuable changes than formatting.

btw I can simply do ctrl+enter on this line in intellij and get 

`List<Map<String, String>> pathSet = new ArrayList<>();`

which for API users is significantly better than `List pathSet = new ArrayList<>();`

Reply:
I agree.
I qualified the touched collections even before reading you comment.

BTW: I did not change all HashMap to Map because some code contains too many ugly casts to HashMap that
won't fail after the changes but the underlying implementation type would be hidden even more.
There is far too many dealing with implementation classes instead of interfaces.

And a global import cleanup should be executed.
Is there some maven target for that in the project?